README.md

```diff
@@ -173,6 +173,56 @@ console.log(await promptWithCalculator("What is 2 + 2?"));
 
 We'll likely explore more specific APIs for tool- and function-calling in the future; follow along in [issue #7](https://github.com/webmachinelearning/prompt-api/issues/7).
 
+### Multimodal inputs
+
+All of the above examples have been of text prompts. Some language models also support other inputs. Our design initially includes the potential to support images and audio clips as inputs. This is done by using objects in the form `{ type: "image", data }` and `{ type: "audio", data }` instead of strings. The `data` values can be the following:
+
+* For image inputs: [`ImageBitmapSource`](https://html.spec.whatwg.org/#imagebitmapsource), i.e. `Blob`, `ImageData`, `ImageBitmap`, `VideoFrame`, `OffscreenCanvas`, `HTMLImageElement`, `SVGImageElement`, `HTMLCanvasElement`, or `HTMLVideoElement` (will get the current frame). Also raw bytes via `BufferSource` (i.e. `ArrayBuffer` or typed arrays).
+
+* For audio inputs: for now, `Blob`, `AudioBuffer`, `HTMLAudioElement`. Also raw bytes via `BufferSource`. Other possibilities we're investigating include `AudioData` and `MediaStream`, but we're not yet sure if those are suitable to represent "clips".
+
+Sessions that will include these inputs need to be created using the `expectedInputTypes` option, to ensure that any necessary downloads are done as part of session creation, and that if the model is not capable of such multimodal prompts, the session creation fails.
+
+A sample of using these APIs:
+
+```js
+const session = await ai.languageModel.create({
+  expectedInputTypes: ["audio", "image"] // "text" is always expected
+});
+
+const referenceImage = await (await fetch("/reference-image.jpeg")).blob();
+const userDrawnImage = document.querySelector("canvas");
+
+const response1 = await session.prompt([
+  "Give a helpful artistic critique of how well the second image matches the first:",
```

Comment:
Since `AILanguageModelPromptInput` is `(DOMString or AILanguageModelPrompt or sequence<AILanguageModelPrompt>)`, the first item of the array parameter should be a `AILanguageModelPrompt`  
```suggestion
  { role: "system", content: "Give a helpful artistic critique of how well the second image matches the first:." },
```
However, I believe there's still something wrong going on as the dictionary `AILanguageModelPrompt` is composed of required `role` and `content` parameters.

Reply:
The WebIDL starts to get complicated.

IIUC, the first parameter is a sequence of `AILanguageModelPromptLine`, which itself can be DOMString ("Give a helpful artistic critique of how well the second image matches the first:") or a `AILanguageModelPromptLineDict`, which itself is a dictionary with required `role` and `content`. Not just `{ type: "image", data: referenceImage }`.

Reply:
Thank you! I think it should be good now, with more comments in the IDL...